httpcore5/src/main/java/org/apache/hc/core5/http/ContentType.java

```diff
@@ -214,6 +236,18 @@ public final class ContentType implements Serializable {
         CONTENT_TYPE_MAP = Collections.unmodifiableMap(map);
     }
 
+    // List of media types that should not include charset
+    private static final List<String> NO_CHARSET_TYPES = Collections.unmodifiableList(Arrays.asList(
```

Comment:
I wouldn't maintain such a like because it can never be exhaustive. We have some predefines and they should carry metadata. If a client creates a custom content type it is his obligation to provide this information, no?